### PR TITLE
Add DS option to disable HMAC packet authentication

### PR DESCRIPTION
**Pull request by SimonWalton-HiFi**
For performance evaluation purposes, make the packet authentication optional with a DS config switch.

Related to Manuscript https://highfidelity.manuscript.com/f/cases/16514/AC-can-drop-assignment-under-high-load

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11416/HighFidelity-Beta-PR13643-d147d17.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11416/HighFidelity-Beta-PR13643-d147d17.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11416/HighFidelity-Beta-Interface-PR13643-d147d17.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11416/HighFidelity-Beta-PR13643-d147d17.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11416/HighFidelity-Beta-Interface-PR13643-d147d17.exe

**Comment by SimonWalton-HiFi** [comment]
## Test Plan

- Run the local sandbox and Interface
- Connect to the domain server web-interface at http://localhost:40100 .
- Go to Settings -> Metaverse/Networking -> Advanced and note Enable Authentication option
- Toggle Enable Authentication option, click on Save & restart
- Test Interface & assignment clients for correctness

The Enable Authentication configuration option is new with this PR. Changing the option's value should enable the Save & restart button. Restarting the DS is necessary for the change to take effect. The value of the option should have no observable effect on the functionality - visiting the local sandbox should work normally, as before the PR. With this PR you should not be allowed to visit an outside domain (e.g. dev-mobile) due to protocol mismatch (for both option values).

enableauth

File: domain-server/resources/describe-settings.json [hunk 1]
```diff
@@ -46,6 +46,14 @@
           "default": "40102",
           "type": "int",
           "advanced": true
+        },
+        {
+          "name": "enable_authentication",
+          "label": "Enable Authentication",
```
[the remark this answers is not in the document]
Comment (SimonWalton-HiFi) [reply domain-server/resources/describe-settings.json]: Done.

File: domain-server/src/DomainServer.cpp [hunk 2]
```diff
@@ -1133,7 +1137,7 @@ void DomainServer::sendDomainListToNode(const SharedNodePointer& node, const Hif
     extendedHeaderStream << node->getUUID();
     extendedHeaderStream << node->getLocalID();
     extendedHeaderStream << node->getPermissions();
-
+    extendedHeaderStream << limitedNodeList->getAuthenticatePackets();
```
[the remark this answers is not in the document]
Comment (SimonWalton-HiFi) [reply domain-server/src/DomainServer.cpp]: I think the logic was that the DomainList is the first packet that a newly-assigned AC receives from the DS with its UUID.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11543/HighFidelity-Beta-PR13643-81c8fe6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11543/HighFidelity-Beta-Interface-PR13643-81c8fe6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11543/HighFidelity-Beta-PR13643-81c8fe6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11543/HighFidelity-Beta-Interface-PR13643-81c8fe6.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11543/HighFidelity-Beta-PR13643-81c8fe6.apk

**Comment by qligmaloney** [comment]
Build this please

**Comment by conklin94122** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11832/HighFidelity-Beta-PR13643-81c8fe6.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11832/HighFidelity-Beta-PR13643-81c8fe6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11832/HighFidelity-Beta-Interface-PR13643-81c8fe6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11832/HighFidelity-Beta-PR13643-81c8fe6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11832/HighFidelity-Beta-Interface-PR13643-81c8fe6.exe

**Comment by qlistephens95** [comment]
Currently Testing

**Comment by qlipdahlin** [comment]
Currently Testing

**Comment by qlipdahlin** [comment]
@conklin94122 PASS

Loaded interface and sandbox, turned off the authentication, saved and restarted, verified sandbox worked fine as local host, another connected and basic usage was still working. Checked logs of interface and AC and no messages of "Packet hash mismatch on ..." were present.

**Pull request merged**